extensions-contrib/momentsketch/src/main/java/org/apache/druid/query/aggregation/momentsketch/aggregator/MomentSketchAggregatorFactory.java

```diff
@@ -0,0 +1,279 @@
+    } else {
+      final ColumnValueSelector<MomentSketchWrapper> selector = metricFactory.makeColumnValueSelector(fieldName);
+      return new MomentSketchMergeBufferAggregator(selector, k, getCompress());
+    }
+  }
+
+  public static final Comparator<MomentSketchWrapper> COMPARATOR = Comparator.comparingDouble(
+      a -> a.getPowerSums()[0]
+  );
+
+  @Override
+  public Comparator getComparator()
+  {
+    return COMPARATOR;
+  }
+
+  @Override
+  public Object combine(Object lhs, Object rhs)
+  {
+    MomentSketchWrapper union = (MomentSketchWrapper) lhs;
+    union.merge((MomentSketchWrapper) rhs);
+    return union;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new MomentSketchMergeAggregatorFactory(name, k, compress);
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass()) {
+      return getCombiningFactory();
+    } else {
+      throw new AggregatorFactoryNotMergeableException(this, other);
+    }
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.<AggregatorFactory>singletonList(
+        new MomentSketchAggregatorFactory(
+            fieldName,
+            fieldName,
+            k,
+            compress
+        )
+    );
+  }
+
+  private MomentSketchWrapper deserializeFromByteArray(byte[] bytes)
+  {
+    return MomentSketchWrapper.fromByteArray(bytes);
+  }
+
+  @Override
+  public Object deserialize(Object serializedSketch)
+  {
+    if (serializedSketch instanceof String) {
+      String str = (String) serializedSketch;
+      return deserializeFromByteArray(Base64.decodeBase64(str.getBytes(Charsets.UTF_8)));
+    } else if (serializedSketch instanceof byte[]) {
+      return deserializeFromByteArray((byte[]) serializedSketch);
+    } else if (serializedSketch instanceof MomentSketchWrapper) {
+      return (MomentSketchWrapper) serializedSketch;
+    }
+    throw new ISE(
+        "Object is not of a type that can be deserialized to a Moments Sketch"
+        + serializedSketch.getClass());
+
+  }
+
+  @Override
+  public Object finalizeComputation(Object object)
+  {
+    return object;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @JsonProperty
+  public int getK()
+  {
+    return k;
+  }
+
+  @JsonProperty
+  public boolean getCompress()
+  {
+    return compress;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return TYPE_NAME;
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return (k + 2) * 8 + 2 * 4 + 8;
```

Comment:
Should use constants like `Integer.BYTES` and also explain each addend in comments